spec/models/order_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+RSpec.describe Order, type: :model do
+  it "is valid with valid attributes" do
+    expect(Order.new).to_not be_valid
+  end
+
+  describe "Scopes" do
+    let!(:user){FactoryBot.create :user, name: "tuongabc"}
```

Comment:
phân biệt let và let!

Reply:
let! định nghĩa và tính toán các biến đó ở đầu giống before 
let chỉ định nghĩa khi nó được gọi lần đầu trong mỗi example

Reply:
Vậy nên dùng let hay let! em ơi

Reply:
em nghĩ là nên dùng let! ạ

Reply:
Em nên hạn chế dùng let! vì nó sẽ luôn tạo ra những giá trị này trước khi vào test case, và có test case sẽ không cần đến nhưng biến như vậy  dẫn đến tốn tài nguyên tốn performance của hệ thông em nhé